### PR TITLE
fix: Select All should only select selectable items

### PR DESCRIPTION
**Pull request by salisdemos**
In dpDataTable:
If the feature "lock item" (`lockCheckboxBy`) is on, these items shouldn't be selected when using select all.

### CODE REVIEW

File: src/components/DpDataTableExtended/DpDataTableExtended.vue [hunk 1]
```diff
@@ -35,18 +35,10 @@
     </dp-sticky-element>
 
     <dp-data-table
-      :has-flyout="hasFlyout"
-      :header-fields="headerFields"
-      :is-expandable="isExpandable"
-      :is-loading="isLoading"
-      :is-resizable="isResizable"
-      :is-selectable="isSelectable"
-      :is-truncatable="isTruncatable"
```
Comment (salisdemos) [inline src/components/DpDataTableExtended/DpDataTableExtended.vue:44]: Just to simplify further adjustments. Not sure if this is a clever move

File: CHANGELOG.md [hunk 2]
```diff
@@ -12,6 +12,7 @@ Since v0.0.10, this Changelog is formatted according to the [Common Changelog][c
 
 ### Changed
 
+- ([#878](https://github.com/demos-europe/demosplan-ui/pull/878)) BREAKING: Prevent "select all" in DpDataTable to select "locked" items. Before all items per page where selected, independed of the the lock state ([@salisdemos](https://github.com/salisdemos))
```
Comment (spiess-demos) [inline CHANGELOG.md:15]: this should go into the "unreleased" section.

File: src/components/DpDataTableExtended/DpDataTableExtended.vue (hunk 1 above)
Comment (spiess-demos) [reply src/components/DpDataTableExtended/DpDataTableExtended.vue:44]: i think it is ok. is there a reason why you define `should-be-selected-items` on its own?

File: CHANGELOG.md (hunk 2 above)
Comment (salisdemos) [reply CHANGELOG.md:15]: omg. thanks!

File: src/components/DpDataTableExtended/DpDataTableExtended.vue (hunk 1 above)
Comment (salisdemos) [reply src/components/DpDataTableExtended/DpDataTableExtended.vue:44]: It was named differently before so I can't just pass it along. And I don't want to refactor it with this change.

File: CHANGELOG.md (hunk 2 above)
Comment (salisdemos) [reply CHANGELOG.md]: d9cc57c50bca011703470b9b5bef024ab53a9e37

**Review by hwiem — CHANGES_REQUESTED** [review]

File: CHANGELOG.md [hunk 3]
```diff
@@ -4,6 +4,10 @@ Since v0.0.10, this Changelog is formatted according to the [Common Changelog][c
 
 ## UNRELEASED
 
+### Changed
+
+- ([#878](https://github.com/demos-europe/demosplan-ui/pull/878)) BREAKING: Prevent "select all" in DpDataTable to select "locked" items. Before all items per page where selected, independed of the the lock state ([@salisdemos](https://github.com/salisdemos))
```
Comment (salisdemos) [inline CHANGELOG.md:9]:
```suggestion
- ([#878](https://github.com/demos-europe/demosplan-ui/pull/878)) BREAKING: Prevent "select all" in DpDataTable from selecting "locked" items. Before, all items per page were selected, independent of the lock state ([@salisdemos](https://github.com/salisdemos))
```

**Review by hwiem — APPROVED** [review]

**Pull request merged**